File: 2. Structural Design Patterns/Adapter/Python/AdapterTemplate.py
# Adapter Pattern – Python Template

from abc import ABC, abstractmethod

class Target(ABC):
    @abstractmethod
    def request(self):
        pass

class Adaptee:
    def specific_request(self):
        return "Special behavior"

class Adapter(Target):
    def __init__(self, adaptee):
        self.adaptee = adaptee
    
    def request(self):
        return f"Adapter: {self.adaptee.specific_request()}"

def main():
    adaptee = Adaptee()
    adapter = Adapter(adaptee)
    print(adapter.request())

if __name__ == "__main__":
    main()

File: 2. Structural Design Patterns/Adapter/Java/AdapterTemplate.java
// Adapter Pattern – Java Template
// Target expects request(); Adaptee has specificRequest(); Adapter bridges them.

interface Target { void request(); }
class Adaptee { void specificRequest() { /* ... */ } }
class Adapter implements Target { private final Adaptee a = new Adaptee(); public void request(){ a.specificRequest(); } }
public class AdapterTemplate { public static void main(String[] args){ Target t = new Adapter(); t.request(); } }
